File: TypeScript/src/questions.ts
export class Questions {

    popQuestions: Array<string> = [];

    public constructor(prefix: string) {

        for (let i = 0; i < 50; i++) {
            this.popQuestions.push(prefix + " Question " + i);
        }
    }

    public askQuestion(): void {
        console.log(this.popQuestions.shift());
    }

}

File: Java/src/main/java/com/adaptionsoft/games/uglytrivia/Questions.java
package com.adaptionsoft.games.uglytrivia;

import java.util.LinkedList;

public class Questions {

    LinkedList popQuestions = new LinkedList();

    public Questions(String prefix) {
        for (int i = 0; i < 50; i++) {
            popQuestions.addLast(prefix + " Question " + i);
        }
    }

    public void askQuestion() {
        System.out.println(popQuestions.removeFirst());
    }

}
